### DISCUSSION TITLE
Pushing sequential packages doesn't show the name of a package currently being deployed

### DISCUSSION THREAD

**Discussion by pznmc** [forcedotcom/cli#2264] [General]
**Is your feature request related to a problem? Please describe.**
Deploying a large number of packages can take a long time. Using `pushPackageDirectoriesSequentially` set to `true` makes all packages to be deployed one by one. When we have more than 40 packages inside `sfdx-project.json`, it's easy to get lost which package is currently being deployed.

**What are you trying to do**
I would like to see the name of a package currently being deployed.

**Describe the solution you'd like**
Instead of seeing the following console output:
```
*** Pushing with SOAP API v54.0 ***
DEPLOY PROGRESS | ████████████████████████████████████████ | 251/251 Components
*** Pushing with SOAP API v54.0 ***
DEPLOY PROGRESS | ████████████████████████████████████████ | 3453/3453 Components
*** Pushing with SOAP API v54.0 ***
DEPLOY PROGRESS | ░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░ | 0/2345 Components
```

I would like to see this for example:
```
*** Pushing pkg-x with SOAP API v54.0 ***
DEPLOY PROGRESS | ████████████████████████████████████████ | 251/251 Components
*** Pushing pkg-y with SOAP API v54.0 ***
DEPLOY PROGRESS | ████████████████████████████████████████ | 3453/3453 Components
*** Pushing pkg-z with SOAP API v54.0 ***
DEPLOY PROGRESS | ░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░ | 0/2345 Components
```

**Describe alternatives you've considered**
I can only wait & count packages, which one is currently deployed, and compare it with order in a `sfdx-project.json` file.

[a comment by an automated account is not shown]

**Comment by cristiand391** [comment]
Marking this as `help wanted` in case you or anyone else wants to add this feature.

How I see this should work:

### 1st scenario: no `pushPackageDirectoriesSequentially` defined, only one pkg:

nothing changes, you know there's only one pkg to deploy.

### 2nd scenario: `pushPackageDirectoriesSequentially` is set to true, multiple pkg will be deployed:

the output should look like what @pznmc suggest so it's clear to the user which pkg is being deployed:

```
*** Pushing pkg-x with SOAP API v54.0 ***
DEPLOY PROGRESS | ████████████████████████████████████████ | 251/251 Components
*** Pushing pkg-y with SOAP API v54.0 ***
DEPLOY PROGRESS | ████████████████████████████████████████ | 3453/3453 Components
*** Pushing pkg-z with SOAP API v54.0 ***
DEPLOY PROGRESS | ░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░ | 0/2345 Components
``` 

## technical details

There's a variable in `force:source:push` to detect if it's doing a sequential deploy, use it to detect which msg to render:
https://github.com/salesforcecli/plugin-source/blob/main/src/commands/force/source/push.ts#L80

you'll need to update the `getVersionMessage` func to accept a pkg name (`string`) as an argument and return the correct msg:
https://github.com/salesforcecli/plugin-source/blob/main/src/commands/force/source/push.ts#L108
https://github.com/salesforcecli/plugin-source/blob/main/src/deployCommand.ts#L234

Also use `this.project.getSfProjectJson().getUniquePackageNames()` to get the pkg names in the order they are defined.

The library that deploys the source returns a "Component Set` of metadata per pkg, the order of the component sets returned will be the same as you have pkgs defined in the project so you can rely on that to assign the each pkg name to a component set here
https://github.com/salesforcecli/plugin-source/blob/main/src/commands/force/source/push.ts#L90

1st `componentSet` corresponds to the 1st pkg returned by `getUniquePackageNames()`, etc.

feel free to post here if you have any questions.

**Comment by cristiand391** [comment]
Closing this because the source commands are deprecated.

`sf project deploy start` doesn't support `pushPackageDirectoriesSequentially`, see this post:
https://github.com/forcedotcom/cli/discussions/2402
